lib/js/src/manager/screen/BaseVoiceCommandManager.js

```diff
@@ -0,0 +1,219 @@
+/*
+* Copyright (c) 2020, Livio, Inc.
+* All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are met:
+*
+* Redistributions of source code must retain the above copyright notice, this
+* list of conditions and the following disclaimer.
+*
+* Redistributions in binary form must reproduce the above copyright notice,
+* this list of conditions and the following
+* disclaimer in the documentation and/or other materials provided with the
+* distribution.
+*
+* Neither the name of the Livio Inc. nor the names of its contributors
+* may be used to endorse or promote products derived from this software
+* without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+* AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { BaseSubManager } from '../BaseSubManager.js';
+import { HMILevel } from '../../rpc/enums/HMILevel.js';
+import { PredefinedWindows } from '../../rpc/enums/PredefinedWindows.js';
+import { FunctionID } from '../../rpc/enums/FunctionID.js';
+import { DeleteCommand } from '../../rpc/messages/DeleteCommand.js';
+import { AddCommand } from '../../rpc/messages/AddCommand.js';
+
+class BaseVoiceCommandManager extends BaseSubManager {
+    /**
+     * @param {LifecycleManager} lifecycleManager
+    */
+    constructor (lifecycleManager) {
+        super(lifecycleManager);
+        this._currentHmiLevel = HMILevel.HMI_NONE;
+        this._taskQueue = [];
+        this._voiceCommands = [];
+        this._voiceCommandIdMin = 1900000000;
+        this._lastVoiceCommandId = this._voiceCommandIdMin;
+        this._isReady = false; // whether the HMI level is appropriate for sending requests
+        this._hmiListener = null;
+        this._commandListener = null;
+        this._isUpdateRunning = false; // prevent _update from being invoked more than once at a time
+        this._addListeners();
+    }
+
+    /**
+     * @return {Promise}
+    */
+    async start () {
+        this._transitionToState(BaseSubManager.READY);
+        super.start();
+    }
+
+    dispose () {
+        this._currentHmiLevel = HMILevel.HMI_NONE;
```

Comment:
Yeah. From what I could find, it being either null or NONE state was treated similarly, so I just keep it non-null to avoid an extra condition for some checks